indra/newview/lldrawpoolmaterials.cpp

```diff
@@ -247,21 +249,28 @@ void LLDrawPoolMaterials::renderDeferred(S32 pass)
         // upload matrix palette to shader
         if (rigged && params.mAvatar.notNull())
         {
-            if (params.mAvatar != lastAvatar)
+            if (params.mAvatar != lastAvatar || params.mSkinInfo != lastSkinInfo)
             {
+                llassert(params.mSkinInfo);
                 const LLVOAvatar::MatrixPaletteCache& mpc = params.mAvatar->updateSkinInfoMatrixPalette(params.mSkinInfo);
                 U32 count = static_cast<U32>(mpc.mMatrixPalette.size());
+                skipLastSkin = !bool(count);
+                lastAvatar = params.mAvatar;
+                lastSkinInfo = params.mSkinInfo;
 
-                if (count == 0)
+                if (!skipLastSkin)
                 {
-                    //skin info not loaded yet, don't render
-                    return;
+                    mShader->uniformMatrix3x4fv(LLViewerShaderMgr::AVATAR_MATRIX,
+                        count,
+                        false,
+                        (GLfloat*)&(mpc.mGLMp[0]));
                 }
+            }
 
-                mShader->uniformMatrix3x4fv(LLViewerShaderMgr::AVATAR_MATRIX,
-                    count,
-                    false,
-                    (GLfloat*)&(mpc.mGLMp[0]));
+            if (skipLastSkin)
+            {
+                //skin info not loaded yet, don't render
+                continue;
```

Comment:
I replaced a return statement with a continue statement. Seems not quite correct to me if one skin fails to load for some reason. Pointing this out in case this is a Chesterton's fence. (@RunitaiLinden wrote this line according to the git logs)